elements/userlevel/dpdkinfo.cc

```diff
@@ -0,0 +1,52 @@
+// -*- mode: c++; c-basic-offset: 4 -*-
+/*
+ * dpdkinfo.{cc,hh} -- library for interfacing with dpdk
+ *
+ * Copyright (c) 2014-2015 Cyril Soldani, University of Liège
+ * Copyright (c) 2015 Tom Barbette, University of Liège
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, subject to the conditions
+ * listed in the Click LICENSE file. These conditions include: you must
+ * preserve this copyright notice, and you cannot mention the copyright
+ * holders in advertising related to the Software without their permission.
+ * The Software is provided WITHOUT ANY WARRANTY, EXPRESS OR IMPLIED. This
+ * notice is a summary of the Click LICENSE file; the license in that file is
+ * legally binding.
+ */
+
+#include <click/config.h>
+#include <click/args.hh>
+#include "dpdkinfo.hh"
+
+CLICK_DECLS
+
+int DpdkInfo::configure(Vector<String> &conf, ErrorHandler *errh) {
+    if (instance) {
+        return errh->error(
+            "There can be only one instance of DpdkInfo!");
+    }
+    instance = this;
+    if (Args(conf, this, errh)
+        .read_p("NB_MBUF", DpdkDevice::NB_MBUF)
```

Comment:
These should all just be `read` rather than `read_p`, since they aren't positional arguments.

I would prefer DPDKInfo, and similar namings; Click generally uses all-caps for acronyms (e.g. CheckTCPHeader).